### PR TITLE
mgr/balancer: eliminate usage of MS infrastructure for upmap mode

### PR DESCRIPTION
**Pull request by xiexingguo**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/balancer/module.py [hunk 1]
```diff
@@ -650,7 +669,10 @@ def serve(self):
                     pool_name_by_id = dict((p['pool'], p['pool_name']) for p in osdmap.dump().get('pools', []))
                     final = [int(p) for p in final]
                     final = [pool_name_by_id[p] for p in final if p in pool_name_by_id]
-                plan = self.plan_create(name, osdmap, final)
+                if self.get_module_option('mode') == 'upmap':
+                    plan = AutoUpmapPlan(name, osdmap, final)
+                else:
+                    plan = self.plan_create(name, osdmap, final)
```
Comment (liewegas) [inline src/pybind/mgr/balancer/module.py]: can we make this a bit cleaner but either making a corresponding CompatSetPlan, and/or making plan_create do the type branch, or possibly even a superclass that defines the common fields?

Comment (neha-ojha) [reply src/pybind/mgr/balancer/module.py]: you could also pass mode to MappingState() and initialize fields based on mode

**Comment by xiexingguo** [comment]
@liewegas Take another look?

**Comment by xiexingguo** [comment]
retest this please

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by xiexingguo** [comment]
@liewegas Ping? I have some more follow-up changes that depend on this.

**Comment by xiexingguo** [comment]
http://pulpito.ceph.com/xxg-2019-12-23_04:13:32-rados-wip-xxg-testing-distro-basic-smithi/

**Pull request merged**